trieste/acquisition/function/function.py

```diff
@@ -325,6 +463,63 @@ def __call__(self, x: TensorType) -> TensorType:
         return ei * augmentation
 
 
+class log_augmented_expected_improvement(AcquisitionFunctionClass):
+    def __init__(self, model: SupportsGetObservationNoise, eta: TensorType):
+        r"""
+        Return the logarithm of Augmented Expected Improvement (AEI) acquisition function
+        for single-objective global optimization under homoscedastic observation noise.
+        Improvement is with respect to the current "best" observation ``eta``, where an
+        improvement moves towards the objective function's minimum and the expectation is
+        calculated with respect to the ``model`` posterior.
+        This shares the same optima as original AEI but expected to be easier to optimize.
+        See
+
+        :param model: The model of the objective function.
+        :param eta: The "best" observation.
+        :return: The expected improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one or a model without homoscedastic observation noise.
+        """
+        self._model = model
+        self._eta = tf.Variable(eta)
+        self._noise_variance = tf.Variable(model.get_observation_noise())
+
+    def update(self, eta: TensorType) -> None:
+        """Update the acquisition function with a new eta value and noise variance."""
+        self._eta.assign(eta)
+        self._noise_variance.assign(self._model.get_observation_noise())
+
+    @tf.function
+    def __call__(self, x: TensorType) -> TensorType:
+        tf.debugging.assert_shapes(
+            [(x, [..., 1, None])],
+            message="This acquisition function only supports batch sizes of one.",
+        )
+        eps = (
+            tf.constant(-1e12, dtype=tf.float64)
+            if x.dtype == tf.float64
+            else tf.constant(-1e6, dtype=tf.float32)
+        )
+        mean, variance = self._model.predict(tf.squeeze(x, -2))
+        sigma = tf.sqrt(tf.clip_by_value(variance, clip_value_min=eps, clip_value_max=x.dtype.max))
+        u = (self._eta - mean) / sigma
+        logei = _log_ei_helper(u) + tf.math.log(sigma)
+
+        """
+        It can be problematic to take the log of original augmentaiton term in AEI
+        since variance + noise_variance  can be numerically equal to noise_variance.
+        Hence, we use numerically stable version
+        """
+
+        log_augmentation = tf.math.log(variance) - tf.math.log(
+            tf.math.sqrt(variance + self._noise_variance)
+        )
+        log_augmentation = log_augmentation - tf.math.log(
+            tf.math.sqrt(variance + self._noise_variance) + tf.math.sqrt(self._noise_variance)
```

Comment:
you could perhaps eliminate duplication with the lines above: `tf.math.sqrt(variance + self._noise_variance)`